rust/blockstore/src/types/writer_options.rs

```diff
@@ -0,0 +1,40 @@
+use uuid::Uuid;
+
+#[derive(Debug, Default, PartialEq, Eq)]
+pub(crate) enum BlockfileWriterMutationOrdering {
+    #[default]
+    Unordered,
+    Ordered,
+}
+
+#[derive(Debug, Default)]
+pub struct BlockfileWriterOptions {
+    pub(crate) mutation_ordering: BlockfileWriterMutationOrdering,
+    pub(crate) fork: Option<Uuid>,
+}
+
+impl BlockfileWriterOptions {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// No guarantees are made about the order of mutations (calls to `.set()` and `.delete()`).
+    pub fn unordered_mutations(mut self) -> Self {
+        self.mutation_ordering = BlockfileWriterMutationOrdering::Unordered;
+        self
+    }
+
+    /// Mutations (calls to `.set()` and `.delete()`) are provided in ascending order of keys. This mode should be preferred when possible as it's more efficient than unordered mutations. Blockfile implementations may panic when in this mode if:
```

Comment:
"This mode should be preferred when possible" seems misleading. If your calling code cannot guarantee order than this is not true.